sync3/caches/user.go

```diff
@@ -263,6 +270,17 @@ func (c *UserCache) OnRegistered(ctx context.Context, _ int64) error {
 				})
 			}
 		}
+
+		// Record when we joined the room. We've just had to scan the history of our
+		// membership in this room to produce joinedRooms above, so we may as well
+		// do this here too.
+		c.roomToDataMu.Lock()
+		urd, ok := c.roomToData[room.RoomID]
+		if !ok {
+			urd = NewUserRoomData()
+		}
+		urd.JoinNID = joinNIDs[room.RoomID]
```

Comment:
You need to remember to set the value, else it won't take effect as you aren't dealing with pointers!
```suggestion
		urd.JoinNID = joinNIDs[room.RoomID]
		c.roomToData[room.RoomID] = urd
```